### PR TITLE
tasks: grouped Celery tasks

### PR DESCRIPTION
**Pull request by jirikuncar**

### CODE REVIEW

File: invenio_oaiserver/query.py [hunk 1]
```diff
@@ -49,10 +49,23 @@ def match(self, record):
...
+    # X          Y          X OR Y
+
+    if spec is None and search_pattern is None:
+        raise StopIteration
+    elif spec is None:
+        query = search_pattern
+    elif search_pattern is None or search_pattern == '':
+        query = '_oai.sets:"{0}"'.format(spec)
+    else:
+        query = '_oai.sets:"{0}" OR {1}'.format(spec, search_pattern)
```
Comment (hachreak) [inline invenio_oaiserver/query.py]: WDYT?
```python
if not spec:
    if not search_pattern:
        raise StopIteration
    query = search_pattern
else:
    query = '_oai.sets:"{0}"'.format(spec)
    if search_pattern:
        query = '{0} OR {1}'.format(query, search_pattern)
```

Comment (jirikuncar) [inline invenio_oaiserver/query.py]: I personally don't line condition nesting.

Comment (hachreak) [inline invenio_oaiserver/query.py]: IMHO in any case it's quite confusing the logic behind

Comment (jirikuncar) [inline invenio_oaiserver/query.py]: That's why I have added https://github.com/inveniosoftware/invenio-oaiserver/pull/52/files#diff-b3c8a68e623bc53b9ab366f36d07dec9R52 :wink:

Comment (hachreak) [inline invenio_oaiserver/query.py]: yes, It was included it in the "logic behind".

**Pull request merged**